### ISSUE TITLE
Bug with RegEx and (Book)marked lines

### ISSUE THREAD

**Issue by rddim** [notepad-plus-plus/notepad-plus-plus#5361]
### Description of the Issue

When I use a regex to mark and bookmark repeated lines and after that use `Remove Unmarked Lines` and `Undo` all the lines (with the uniques one) between 1st and last bookmarked lines are also marked.
### Steps to Reproduce the Issue

1. In a text file type:
```
aaa
bbb
ccc
aaa
aaa
ddd
eee
bbb
bbb
```
2. Go to `Search` > `Mark...` 
3. In `Find what:` type `^(.*)$\s+(?=(^.*$\s+)*\1)` or `^(.*?)$\s+?^(?=.*^\1$)`
4. Check the `Bookmark line`, `Regular expression` and `. matches newline`
5. Click `Mark All`
6. Go to `Search` > `Bookmark` > `Remove Unmarked Lines`
7. Click on `Undo` icon or `Edit` > `Undo`

### Expected Behavior

After the step `7` I expect to seen the result that is after the step `5`
### Actual Behavior

The unique lines between 1st and last bookmarked lines are also marked.
![npp_regex_mark_bookmark](https://user-images.githubusercontent.com/8998791/53645078-c740d880-3c40-11e9-8a6d-14596135d199.gif)

### Debug Information

Notepad++ v7.6.3   (64-bit)
Build time : Jan 27 2019 - 17:16:47
Path : D:\Install\Office Programs\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : AutoSave.dll ComparePlugin.dll VisualStudioLineCopy.dll 

**Edit:**
1. If I use `Normal` search mode and `Find what:` `aaa` or `bbb` the result is normal with or without bookmarks
2. If I use step `3` but type `aaa` or `bbb` the result is also normal with or without bookmarks

**Comment by sasumner** [comment]
To my knowledge, Undo refers to textual changes only and not bookmarking (or other actions).  You are certainly free to complain about it not working well with bookmarking.

**Comment by rddim** [comment]
OK, I make another test:
After step `6` and `Clear All Bookmarks` the result is the same after `Undo`

**Comment by sasumner** [comment]
Unlike @MetaChuh , I can't follow the reproduction steps well enough to make a determination.  @rddim please provide a clearer set of steps, showing what you expect at each step.  If this doesn't happen in a few days, I'm going to go ahead and close this issue.

**Comment by rddim** [comment]
@sasumner I am away from home and my PC now. In the middle of the next week I will be available. 

На 7 август 2019 г. 21:45:22 GMT+03:00, sasumner <notifications@github.com> написа:
>Unlike @MetaChuh , I can't follow the reproduction steps well enough to
>make a determination.  @rddim please provide a clearer set of steps,
>showing what you expect at each step.  If this doesn't happen in a few
>days, I'm going to go ahead and close this issue.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/notepad-plus-plus/notepad-plus-plus/issues/5361#issuecomment-519222050

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sasumner** [comment]
@rddim That's fine, I will give until 19 Aug 2019 before auto-closure.

**Comment by rddim** [comment]
@sasumner 

After following all the 7 steps I expect **only** lines numbers `1` `2` `4` and `8` to be marked and bookmarked.

**Comment by sasumner** [comment]
@rddim Okay, I think I see it now. Sorry for not seeing it earlier and thank you for the clarification. 

The complaint is that after the 7 steps to reproduce there is too much (red)marking:  Lines 1-8 are all completely (red)marked when the contention is that only lines 1,2,4,8 should be.  There is no  complaint about how the bookmarks end up after the undo (at least I don't think there is).

[The source of my confusion:  The term "mark" is unfortunate (for me at least), because I get it confused between the (red)_marking_ of text and the (book)_marking_ of lines.  I would much rather have "marking" be officially called "red-marking", but of course the color can be changed by the user -- red is only the default -- so that's probably not a good idea.]

**Comment by sasumner** [comment]
So I still think that "Undo refers to textual changes only...".  Consider an initial state:

![image](https://user-images.githubusercontent.com/30118311/70239452-5e3f4180-1739-11ea-99ad-0b5eeb764e5a.png)

Then make this selection:

![image](https://user-images.githubusercontent.com/30118311/70239547-7d3dd380-1739-11ea-9a25-450534a759bd.png)

Press Delete key to get:

![image](https://user-images.githubusercontent.com/30118311/70239597-96468480-1739-11ea-8bc7-bc8c7922a217.png)

Press ctrl+z (Undo) to get:

![image](https://user-images.githubusercontent.com/30118311/70239639-ad857200-1739-11ea-8ca0-ee7524d46888.png)

Note lack of restoration of redmarking on line 4.

This works similarly for "styling" (similar sequence):

![image](https://user-images.githubusercontent.com/30118311/70239764-e7567880-1739-11ea-8dc2-b534a174c267.png)

![image](https://user-images.githubusercontent.com/30118311/70239797-f9d0b200-1739-11ea-91cd-82fc5729f3f4.png)

![image](https://user-images.githubusercontent.com/30118311/70239819-07863780-173a-11ea-8411-feaad1c50bef.png)

![image](https://user-images.githubusercontent.com/30118311/70239851-153bbd00-173a-11ea-98ed-3dae9a8eeed3.png)

In short, I don't think we can call this a "bug".  If it is a "feature request" then maybe I feel better about that. :-)

**Comment by rddim** [comment]
Then you can close this issue...
